### ISSUE TITLE
Use of CSS `background` property confuses automated accessibility testing tools

### ISSUE THREAD

**Issue by stevenwaterman** [nhsuk/nhsuk-frontend#676]
# Bug Report

## What is the issue?

Some components, including [breadcrumbs](https://service-manual.nhs.uk/design-system/components/breadcrumbs) and [expanders](https://service-manual.nhs.uk/design-system/components/expander) use the CSS `background` property to display icons on elements that also contain text.

This causes warnings or failures in some automated accessibility testing tools, as they detect that a textbox has a background applied and therefore the text may not satisfy contrast requirements. The issue isn't that they *don't* meet contrast requirements, it's that the CSS is used in a non-semantic way violating best practices. The tool assumes that you are following best practices and using the background for a background rather than an icon, which could be an accesibility concern. The tooling isn't (nor should anyone expect it to be) clever enough to detect that the background doesn't actually touch the text.

This is easily solvable by moving the `+` icon on the expander into the `:before` psuedo-class, in the same way that the `details` component does. Similar solutions using `:before` and `:after` would work for the breadcrumbs component too.

See this ugly hack which can be applied to recreate the previous expander pixel-perfect without the warnings using `:before`:

```
.nhsuk-expander {
  .nhsuk-details__summary-text {
    background: none;
  }
  
  .nhsuk-details__summary:before {
    background: url("data:image/svg+xml,%3Csvg class='nhsuk-icon nhsuk-icon__plus' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' aria-hidden='true'%3E%3Ccircle cx='12' cy='12' r='10' fill='%23005eb8'%3E%3C/circle%3E%3Cpath fill='none' stroke='%23fff' stroke-linecap='round' stroke-miterlimit='10' stroke-width='2' d='M12 8v8M8 12h8'%3E%3C/path%3E%3C/svg%3E%0A") no-repeat;
    border: none!important;
    display: block!important;
    clip-path: none;
    -webkit-clip-path: none;
    width: 32px;
    height: 32px;
    left: 22px;
  }
}
```

## What steps are required to reproduce the issue?

Open any page containing an expander such as [the service manual docs](https://service-manual.nhs.uk/design-system/components/expander) and run [axe](https://www.deque.com/axe/)
![Screenshot 2020-12-31 at 14 47 25](https://user-images.githubusercontent.com/23333247/103414741-22162480-4b77-11eb-911a-6a1b46640c08.png)

## What was the environment where this issue occurred?

This is an issue with the structure of the HTML so this is probably redundant but:

* Device: Macbook Pro 2019
* Operating System: MacOS
* Browser: Firefox
* Browser version: 84
* NHS.UK frontend package version: Latest (I assume, whatever your docs are using)
* Node version: NA
* npm version: NA

**Comment by AdamChrimes** [comment]
Thanks for opening this issue @stevenwaterman, I have raised this with our accessibility specialist and we are looking into it. We'll post back any findings here soon.

**Comment by mcheung-nhs** [comment]
Hi @stevenwaterman, as you say, the issue raised by AXE isn't a contrast failure as such, so this can be ignored. However, I was wondering if you could you expand (no pun intended!) on why you say the use of the CSS background to display an icon is not best practice? Thanks.

**Comment by stevenwaterman** [comment]
As I mentioned in my original comment, despite the fact that it's a warning it's still the kind of thing I would want to fix so as to not [normalise the presence of many accesibility warnings](https://en.wikipedia.org/wiki/Normalization_of_deviance). If it's not fixed at a library level then we're going to have to use the hacky workaround I posted above which is going to be a pain to maintain.

Using CSS background to show an icon isn't *that* weird, but the issue here is the fact that it's the background of a text-containing element. If it was just the background of an empty div, that would be fairly normal.

Using background in this manner is bad practice because it's semantically counterintuitive, and breaks a lot of the assumptions that we make about HTML & CSS. If I look at the button on that expander, I don't expect it to be a CSS background over the whole card, but it is. When I see a CSS background used without having looked at the page, I don't expect it to just be used for a single non-repeating icon. It also complicates things, meaning that the card text needs a huge left padding applied to it so it shifts over out of the way of the icon.

Axe is getting confused *because* this breaks those conventions and best practices. It's not so much that there's a technical issue, it's just non-idiomatic. It's as if I wrote a while loop as `for(; condition; )` - like it'd work, but I wouldn't be surprised if automated tools hated it, and the compiler couldn't do any optimisations.

There are a number of more semantic options:
* Using an `svg` element, which is the most semantically meaningful
* Using the ::before psuedo-class, like in my example above (the library *already* does this for the normal details component)
* Explicitly adding an empty div with the icon applied as a background

**Comment by andymantell** [comment]
I can see a couple of sides to this. On the one hand it seems like a bit of a shame to change the way something is built just because aXe or other tools can't handle it. I don't quite agree that using a background image in this way is not semantic - it's semantically inert and I can't see any genuine accessibility issues with the way this is currently built.

_However_, using a `::before` as you have mentioned doesn't have any downsides relative to the current approach and you've largely written the code already, and is consistent with the existing details code as you say. Would you be able to open a pull request with this change that we can review?

**Comment by stevenwaterman** [comment]
It's been a while, sorry about that! PR opened, link above ^
